bigquery-storage/api/BigQueryStorage.Samples.Tests/AppendRowsPendingTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Google.Api.Gax;
+using Google.Apis.Bigquery.v2.Data;
+using Google.Cloud.BigQuery.V2;
+using Xunit;
+
+[Collection(nameof(BigQueryStorageFixture))]
+public class AppendRowsPendingTests
+{
+    private readonly BigQueryStorageFixture _fixture;
+
+    private readonly AppendRowsPendingSample _sample;
+
+    private Dictionary<long, (long, string)> _testData;
+
+    public AppendRowsPendingTests(BigQueryStorageFixture bigQueryStorageFixture)
+    {
+        _fixture = bigQueryStorageFixture;
+        _sample = new AppendRowsPendingSample();
+        PopulateTestData();
+    }
+
+    [Fact]
+    public async Task TestAppendRows()
+    {
+        // Arrange: Ensure dataset and table exists
+        string datasetId = $"dataset{Guid.NewGuid().ToString().Substring(0, 18).Replace('-', '_')}"; // - is not allowed in name
+        string tableName = "append_row_pending";
+        TableSchema tableSchema = new TableSchemaBuilder()
+        {
+            { "customer_number", BigQueryDbType.Int64 },
+            { "customer_name", BigQueryDbType.String }
+        }.Build();
+
+        _fixture.EnsureDataSetAndTableExists(datasetId, tableName, tableSchema);
+
+        // Act: Append Rows in new database
+        await _sample.AppendRowsPendingAsync(_fixture.ProjectId, datasetId, tableName);
+
+        // Get the inserted rows in table.
+        PagedEnumerable<TableDataList, BigQueryRow> insertedData = await _fixture.GetTableDataAsync(datasetId, tableName);
+
+        int count = 1;
```

Comment:
It looks like really your test data is effectively a list, not a dictionary, if the key is just the order in which the rows are expected to appear. The code could be significantly simpler that way.

You never use Item1 either, as far as I can see - so the field could just be:

```csharp
List<string> _testData = new List<string> { "Alice", "Bob", "Charles" };
```

Reply:
Right Jon. Changed it to list.